urlb57eb64056_jacadenac_ISI_Othello_UN_tgz-pJ8-othello_OthelloJ8/src/ui/TableroUI.java

```diff
@@ -0,0 +1,488 @@
+import java.awt.Container;
+import java.awt.Font;
+import java.awt.GridBagConstraints;
+import java.awt.GridBagLayout;
+import java.awt.GridLayout;
+import java.awt.HeadlessException;
+import java.awt.Image;
+import java.awt.Point;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import javax.swing.BorderFactory;
+import javax.swing.ButtonGroup;
+import javax.swing.ImageIcon;
+import javax.swing.JButton;
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+import javax.swing.JMenu;
+import javax.swing.JMenuBar;
+import javax.swing.JMenuItem;
+import javax.swing.JOptionPane;
+import javax.swing.JPanel;
+import javax.swing.JRadioButtonMenuItem;
+
+import ui.FabricaImagenCasilla.TipoCasilla;
+import utilidades.Transform;
+import othello.Othello;
+
+
+public final class TableroUI extends JFrame {
+	private static final long serialVersionUID = 1L;
+	public static final int HUM_HUM = 0;
+	public static final int HUM_ROB = 1;
+	public static final int ROB_ROB = 2;
+	
+	public static int largoTablero = Tablero.TABLERO_LARGO;
+	public static int anchoTablero = Tablero.TABLERO_LARGO;
+
+	private final Image LOGO = new ImageIcon(getClass().getResource("/ui/imagenes/logo.jpg")).getImage();
+	private static int oponentes = HUM_HUM;
+	private static Jugador humano = new Jugador(EstadoCasilla.BLACK);
+	private List<ComponenteImagen> casillas;
+	private JPanel tablero;
+	private JLabel estadisticasBlanco;
+	private JLabel estadisticasNegro;
+	private JLabel mostrarTurno;
+	private JMenuItem juegoNuevo;
+	private JMenuItem guardarJuego;
+	private JMenuItem abrirJuego;
+	private JRadioButtonMenuItem[] tipoTableroButton;
+	private JRadioButtonMenuItem[] tamTableroButton;
+	private Tablero tableroLogica;
+	private JPanel barraEstado;
+	private JPanel barraBoton;
+	private Container pane = this.getContentPane();
+	public JButton boton;
+	TipoTablero tipoTablero;
+
+	
+	public TableroUI() throws HeadlessException {
+		casillas = new ArrayList<ComponenteImagen>(largoTablero*anchoTablero);
+		tipoTablero=TipoTablero.CLASICO;
+		iniciarComponentes();
+		this.pack();
+		this.setLocationRelativeTo(null);
+		this.setDefaultCloseOperation(EXIT_ON_CLOSE);
+		this.setTitle(Othello.class.getSimpleName());
+		this.setIconImage(LOGO);
+		this.setResizable(false);
+	}
+	
+	public TableroUI(int largo, int ancho, Tablero tableroLogica, TipoTablero tipoTablero) throws HeadlessException {
+		largoTablero = largo;
+		anchoTablero = ancho;
+		this.tableroLogica = tableroLogica;
+		this.tipoTablero=tipoTablero;
+		casillas = new ArrayList<ComponenteImagen>(tableroLogica.obtenerTablero().size());
+		iniciarComponentes();
+		this.pack();
+		this.setLocationRelativeTo(null);
+		this.setDefaultCloseOperation(EXIT_ON_CLOSE);
+		this.setTitle(Othello.class.getSimpleName());
+		this.setIconImage(LOGO);
+		this.setResizable(false);
+	}
+	
+	
+	private void iniciarComponentes(){
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 144 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 99% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 100-366 into a separate method.